.github/workflows/CI.yml

```diff
@@ -0,0 +1,30 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Comment:
Shouldn't we also have builds working on mac, maybe windows ? (@keithdoggett I don't the app state on windows support)

Reply:
yeah I think we should include those. I've used it on windows before.

Reply:
I've updated, but specs are failing on Windows. I set them as allowed failures for now. Ideally someone else can tackle getting things working on Windows after this is merged

Reply:
Yeah we can deal with fixing that once it's merged. Are you able to add `macos-latest` as well? It may be easier to break the task up into different jobs for each OS since installing libgeos will be different for each.

Something like this:
```yml
name: CI

on:
  push:
    branches:
      - master
      - github_actions
  pull_request:

jobs:
  ubuntu:
    strategy:
      fail-fast: false
      matrix:
        ruby: [ruby-head, 3.0, 2.7, 2.6, 2.5, jruby]
    runs-on: ubuntu-latest
    continue-on-error: ${{ matrix.ruby == 'ruby-head' }}
    steps:
      - uses: actions/checkout@v2
      - uses: ruby/setup-ruby@v1
        with:
          ruby-version: ${{ matrix.ruby }}
          bundler-cache: true
      - run: |
          sudo apt-get install libgeos-dev -y
      - run: |
          bundle exec rake
  macos:
    strategy:
      fail-fast: false
      matrix:
        ruby: [ruby-head, 3.0, 2.7, 2.6, 2.5, jruby]
    runs-on: macos-latest
    continue-on-error: true
    steps:
      - uses: actions/checkout@v2
      - uses: ruby/setup-ruby@v1
        with:
          ruby-version: ${{ matrix.ruby }}
          bundler-cache: true
      - run: |
          brew install geos
      - run: |
          bundle exec rake
  windows:
    strategy:
      fail-fast: false
      matrix:
        ruby: [ruby-head, 3.0, 2.7, 2.6, 2.5, jruby]
    runs-on: windows-latest
    continue-on-error: true
    steps:
      - uses: actions/checkout@v2
      - uses: ruby/setup-ruby@v1
        with:
          ruby-version: ${{ matrix.ruby }}
          bundler-cache: true
      # figure out how to install on windows. Probably need mingw
      - run: |
          sudo apt-get install libgeos-dev -y
      - run: |
          bundle exec rake
```
I haven't tested this out, but I think something like this would make it easier to figure out how to build for different operating systems.

Reply:
Updated to add macOS too. Also set up scaffolding for OS-specific setup (Windows is blank right now).